### PR TITLE
Fix GH PR discussion snippets filename resolution

### LINKED ISSUE #5728: "cannot read property 'title' of null" when a PR description page contains resolved comments
1. Visit a PR description page that contains resolved comments (eg https://github.com/sourcegraph/sourcegraph/pull/5712)
2. Witness error in console:

```
Uncaught TypeError: Cannot read property 'title' of null
    at getDeltaFileName (util.tsx:24)
    at Object.resolveDiffFileInfo [as resolveFileInfo] (file_info.ts:8)
    at MergeMapSubscriber.project (code_intelligence.tsx:551)
    at MergeMapSubscriber._tryNext (mergeMap.ts:141)
    at MergeMapSubscriber._next (mergeMap.ts:132)
    at MergeMapSubscriber.Subscriber.next (Subscriber.ts:100)
    at IntersectionObserver.<anonymous> (views.ts:135)
```
3. Click on the "files" tab (no hard reload)
4. Witness no code intelligence.

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #5728

Fixes filename resolution (and thus code intelligence) for commented snippets on the GitHub.com PR discussion page: these snippets no longer include a `.file-info` element.

Updates the code-view.html fixtures for github.com to reflect the new DOM, and adds unit tests for `getDeltaFilename()`ensuring that filenames are still correctly resolved for all other types of code views.

Also makes sure to return `diffConversationCodeView` for embedded snippets, so that we don't attempt to mount the code view toolbar.

### CODE REVIEW

**Comment by lguychard** [comment]
@slimsag disregard this request for review. It is due to the following line in CODEOWNERS:

```
**/*discussion* @slimsag
```

Sure enough, it matches all the `pull-request-discussion` fixtures 😅

File: browser/src/libs/github/util.tsx [hunk 1]
```diff
@@ -18,15 +18,28 @@ export function getFileContainers(): HTMLCollectionOf<HTMLElement> {
...
-    const info = container.querySelector('.file-info') as HTMLElement
-
-    if (info.title) {
-        // for PR conversation snippets
-        return getPathNamesFromElement(info)
+export function getDeltaFileName(container: HTMLElement): { headFilePath: string; baseFilePath?: string } {
+    const fileInfoElement = container.querySelector<HTMLElement>('.file-info')
+    if (fileInfoElement) {
+        if (fileInfoElement.tagName === 'A') {
+            // for PR conversation snippets on GHE, where the .file-info element
```
Comment (felixfbecker) [inline browser/src/libs/github/util.tsx]: For my understanding: PR conversation snippet refers to a code snippet from embedding a link _inside_ a comment, or to a snippet of the line a review comment was added to?

File: browser/src/libs/github/util.tsx [hunk 2]
```diff
@@ -18,15 +18,28 @@ export function getFileContainers(): HTMLCollectionOf<HTMLElement> {
  * getDeltaFileName returns the path of the file container. It assumes
  * the file container is for a diff (i.e. a commit or pull request view).
  */
-export function getDeltaFileName(container: HTMLElement): { headFilePath: string; baseFilePath: string | undefined } {
-    const info = container.querySelector('.file-info') as HTMLElement
-
-    if (info.title) {
-        // for PR conversation snippets
-        return getPathNamesFromElement(info)
+export function getDeltaFileName(container: HTMLElement): { headFilePath: string; baseFilePath?: string } {
```
Comment (felixfbecker) [inline browser/src/libs/github/util.tsx]: What do you think about renaming `Delta`->`Diff` (and in the error message too)? I find the term "delta" very confusing in this context

File: browser/src/libs/github/util.test.ts [hunk 3]
```diff
@@ -60,3 +61,68 @@ describe('util', () => {
...
+            // TODO add examples with renamed files and check that getDeltaFilename() doesn't return
+            // identical headFilePath & baseFilePath
+            expect(getDeltaFileName(container)).toStrictEqual({
+                baseFilePath: expectedFilePath,
+                headFilePath: expectedFilePath,
+            })
+        })
+    }
+    for (const gitHubVersion of ['github.com', 'ghe-2.14.11'] as const) {
+        describe(`${gitHubVersion}`, () => {
```
Comment (felixfbecker) [inline browser/src/libs/github/util.test.ts]: nit: in other tests we make them a bit more human-readable with a prefix and by humanizing casing

File: browser/src/libs/github/util.tsx (hunk 1 above)
Comment (lguychard) [reply browser/src/libs/github/util.tsx]: A snippet of the line a review comment was added to:

image

File: browser/src/libs/github/util.tsx (hunk 2 above)
Comment (lguychard) [reply browser/src/libs/github/util.tsx]: Good call 👍

File: browser/src/libs/github/__snapshots__/dom_functions.test.ts.snap [hunk 4]
```diff
@@ -476,30 +476,34 @@ Object {
...
 `;
 
-exports[`GitHub DOM functions diffDomFunctions github.com Pull Request Discussion page Refined Github line number 62 in head diff part getCodeElementFromLineNumber() should return the right code element given the line number 1`] = `
+exports[`GitHub DOM functions diffDomFunctions github.com Pull Request Discussion page Refined Github line number 13 in head diff part getCodeElementFromLineNumber() should return the right code element given the line number 1`] = `
 Object {
-  "content": "if len(parsedUrl.Opaque) > 0 {",
-  "selector": "TR:nth-child(4) > TD.blob-code.blob-code-addition > SPAN.blob-code-inner.blob-code-marker-addition",
+  "content": "export
+                                class TestResourceManager
+                                {",
```
[the remark this answers is not in the document]
Comment (lguychard) [reply browser/src/libs/github/__snapshots__/dom_functions.test.ts.snap]: Fixed.

[the remark this answers is not in the document]
Comment (felixfbecker) [reply browser/src/libs/github/__snapshots__/dom_functions.test.ts.snap]: I guess this was caused by Prettier-formatting the HTML?

**Comment by felixfbecker** [comment]
Thanks for fixing 🙏

**Pull request merged**